src/tests/examples.rs

```diff
@@ -108,6 +110,23 @@ fn test_file(
             compiled_circuit
                 .generate_witness(public_inputs.clone(), private_inputs.clone())
                 .unwrap();
+
+            // check the ASM
+            if compiled_circuit.circuit.backend.len() < 100 {
```

Comment:
btw not a fan of this `len()` function. IMO it should have a better name (like `num_constraints`).